packages/core/src/utils/settings.ts

```diff
@@ -40,14 +64,25 @@ export function updateSettings(updates: ClackSettings) {
 
 		switch (key) {
 			case 'aliases': {
-				for (const alias in value) {
-					if (!Object.hasOwn(value, alias)) continue;
+				const aliases = value as Record<string, Action>;
```

Comment:
you don't need to cast this as it could actually be `undefined`

so you should probably just:

```ts
if (value === undefined) {
  break;
}
```

then typescript will know it is non-nullish later

Reply:
if i don't cast it i get red squiggly 
![image](https://github.com/user-attachments/assets/469841ba-ad40-41b3-b517-70d46cfa233a)

Reply:
the messages cast does work without the assertion 🫡

Reply:
i just read about the typeguard you want me to avoid the typecasting 🫣

Reply:
specifically you need to handle when it is `undefined`, because it can be

typescript is presumably telling you that in the error in your screenshot